firedrake/mesh.py

```diff
@@ -3094,6 +3094,7 @@ def Mesh(meshfile, **kwargs):
     if netgen and isinstance(meshfile, netgen.libngpy._meshing.Mesh):
         netgen_firedrake_mesh.createFromTopology(topology, name=plex.getName(), comm=user_comm)
         mesh = netgen_firedrake_mesh.firedrakeMesh
+        mesh.name = name
```

Comment:
This code looks a bit funny:
* We create a `mesh` on line 3093 which gets discarded - put inside an `else` block?
* We already pass `name` to `createFromTopology` on line 3095. It seems a bit funny to be setting it twice. Should this fix instead go into `ngsPETSc`?